stdlib/LinearAlgebra/test/tridiag.jl

```diff
@@ -291,8 +312,12 @@ end
                     @test_throws DimensionMismatch Tldlt\rand(elty,n+1)
                     @test size(Tldlt) == size(Ts)
                     if elty <: AbstractFloat
+                        @test LinearAlgebra.LDLt{elty,SymTridiagonal{elty,Vector{elty}}}(Tldlt) === Tldlt
+                        @test LinearAlgebra.LDLt{elty}(Tldlt) === Tldlt
+                        @test typeof(convert(LinearAlgebra.LDLt{Float32,Matrix{Float32}},Tldlt)) ==
+                            LinearAlgebra.LDLt{Float32,Matrix{Float32}}
                         @test typeof(convert(LinearAlgebra.LDLt{Float32},Tldlt)) ==
-                            LinearAlgebra.LDLt{Float32,SymTridiagonal{elty,Vector{elty}}}
+                            LinearAlgebra.LDLt{Float32,SymTridiagonal{Float32,Vector{Float32}}}
```

Comment:
I found the old result here surprising; for example, when passed a `SymTridiagonal{Float64,Vector{Float64}}` this conversion created a `LDLt{Float32,SymTridiagonal{Float64,Vector{Float64}}}`, meaning it claimed Float32 precision but internally it was stored with Float64 precision. By changing the type declaration from `LDLt{T,S<:AbstractMatrix}` to `LDLt{T,S<:AbstractMatrix{T}}` I was forced to change this outcome to `LDLt{Float32,SymTridiagonal{Float32,Vector{Float32}}}`. Was the previous outcome actually desired? Or was it essentially an oversight?

Reply:
I'm pretty sure that must have been an oversight.

Reply:
I think these definitions and tests predates the time when `LDLt{T,S<:AbstractMatrix{T}}` was possible.

Reply:
I'm not surprised. I have a ton of constructors that need fixing littered through the package landscape and my lab's own internal code.